src/java.base/share/classes/java/lang/Double.java

```diff
@@ -143,6 +143,55 @@
  * elements of a {@link java.util.SortedSet SortedSet} or as keys of a
  * {@link java.util.SortedMap SortedMap}.
  *
+ * <p>While {@code ==} is not an equivalence relation, several useful
+ * equivalence relations can be defined over floating-point
+ * values. Comparing numerical equality to various kinds of
+ * equivalence:
+ *
+ * <dl>
+ * <dt><i>numerical equality</i> ({@code ==} operator): (<em>Not</em>
+ * an equivalence relation)</dt>
+ * <dd>Two floating-point values represent the same extended real
+ * number. The extended real numbers are the real numbers augmented
+ * with positive infinity and negative infinity. Under numerical
+ * equality, {@code +0.0} and {@code -0.0} are equal since they both
+ * map to the same real value, 0. A NaN does not map to any real
+ * number and is not equal to any value, including itself.
+ * </dd>
+ *
+ * <dt><i>bit-wise equivalence</i>:</dt>
+ * <dd>The bits of the two floating-point values are the same. This
+ * equivalence relation for {@code double} values {@code a} and {@code b} is
+ * implemented by the expression
+ * <br>{@code Double.doubleTo}<code><b>Raw</b></code>{@code LongBits(a) == Double.doubleTo}<code><b>Raw</b></code>{@code LongBits(b)}<br>
+ * Under this relation, {@code +0.0} and {@code -0.0} are distinguished
+ * from each other and every bit pattern encoding a NaN is distinguished
+ * from every other bit pattern encoding a NaN.
+ * </dd>
+ *
+ * <dt><i>representation equivalence</i>:</dt>
+ * <dd>The two floating-point values represent the the same IEEE 754
+ * <i>datum</i>. In particular, for {@linkplain #isFinite(double)
+ * finite} values, the sign, {@linkplain Math#getExponent(double)
+ * exponent}, and significand components of the
+ * floating-point values are the same. Under this relation:
+ * <ul>
+ * <li> {@code +0.0} and {@code -0.0} are distinguished from each other.
+ * <li> every bit pattern encoding a NaN is considered equivalent to each other
+ * <li> an infinite value is equivalent to an infinite value of the same sign
```

Comment:
Seems like this line on infinities could be reworded. I wouldn't quibble over this except that I had to read it several times to figure out what it meant. The statement on NaN is universally quantified, whereas the statement on infinite values starts off sounding like an existential quantifier. Possibly: "all infinite values of the same sign are considered equivalent to each other."

Reply:
Update as suggested and added some cross-links in from BigDecimal; thanks.

Reply:
PS To further spell things out, I added an additional trailing paragraph:

"For two binary floating-point values a and b, if neither of a and b is zero or NaN, then the three relations numerical equality, bit-wise equivalence, and representation equivalence of a and b have the same true/false value. In other words, for binary floating-point values, the three relations only differ if at least one argument is zero or NaN."

Reply:
The additional paragraph sounds fine.